main/src/com/google/refine/importing/ImportingUtilities.java

```diff
@@ -255,107 +281,26 @@ public void update(long bytesRead, long contentLength, int itemCount) {
                     
                 } else if (name.equals("download")) {
                     String urlString = Streams.asString(stream);
-                    URL url = new URL(urlString);
-                    
-                    JSONObject fileRecord = new JSONObject();
-                    JSONUtilities.safePut(fileRecord, "origin", "download");
-                    JSONUtilities.safePut(fileRecord, "url", urlString);
+                    download(rawDataDir, retrievalRecord, progress, fileRecords, update, urlString);
+                } else if (name.equals("data-package")) {
```

Comment:
we could flip this around to avoid an NPE and it will just return false ... but maybe throwing that NPE is better for debugging for now ? https://www.codacy.com/app/OpenRefine/OpenRefine/file/13416576553/issues/source?bid=6129680&fileBranchId=6129680#l285

Reply:
I think we should put name ahead if we are sure the name supposed to have a value flip it around silently hide the issue and redirect the execution to last default case.

Reply:
yeah, my thoughts also, better to not hide in this case.  Also... isn't the name without hypen per the spec ?  
A Data Package descriptor MUST be a valid JSON object. (JSON is defined in RFC 4627). When available as a file it MUST be named **datapackage.json** and it MUST be placed in the top-level directory (relative to any other resources provided as part of the data package).

Reply:
The name is just a key for internal exchange(from the javascript), not the file name itself